File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, ar: List[int]) -> int:
        res = 1
        n = len(ar)
        for i in range(1,n):
            if(ar[i]!=ar[i-1]):
                ar[res] = ar[i]
                res += 1
        return res
        
        

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int j = 0;
        map<int,int> freq;

        for(int i=0;i<nums.size();i++)
        {
            if(freq[nums[i]]==0)
            {
                freq[nums[i]]++;
                nums[j++]=nums[i];
            }
            
        }
        return j;
    }
};